packages/android-client/FishjamClient/src/main/java/com/fishjamcloud/client/ReconnectionManager.kt

```diff
@@ -0,0 +1,78 @@
+package com.fishjamcloud.client
+
+import com.fishjamcloud.client.models.ReconnectConfig
+import timber.log.Timber
+import java.util.Timer
+import kotlin.concurrent.schedule
+
+interface ReconnectionManagerListener {
+  fun onReconnectionStarted() {
+    Timber.i("Reconnection started")
+  }
+
+  fun onReconnected() {
+    Timber.i("Reconnected successfully")
+  }
+
+  fun onReconnectionRetriesLimitReached() {
+    Timber.e("Reconnection retries limit reached")
+  }
+}
+
+enum class ReconnectionStatus {
+  IDLE,
+  RECONNECTING,
+  ERROR,
+  WAITING
+}
+
+internal class ReconnectionManager(
+  private var reconnectConfig: ReconnectConfig = ReconnectConfig(),
+  private val connect: () -> Unit
+) {
+  private val listeners = mutableListOf<ReconnectionManagerListener>()
```

Comment:
No particular reason, just that we have a list of listeners everywhere now, because it's easier to set up tests that way. But I can change it to a variable if there is an issue with a list.

Reply:
let it be